### PR TITLE
Add multihost testing

### LINKED ISSUE #164: RFE: Support multi-host testing
Testing Farm supports [tmt's multihost testing](https://tmt.readthedocs.io/en/stable/guide.html#multihost-testing). According to the [TF multihost](https://docs.testing-farm.io/Testing%20Farm/0.1/test-request.html#multihost-testing) documentation, it is disabled by default and for enabling the pipeline type setting must be set:
```json
{
    "settings": {
        "pipeline": {
            "type": "tmt-multihost"
        }
    }
}
```
Or using the testing farm CLI with `--pipeline-type tmt-multihost`. 

As far as I've seen, it is not yet possible to pass this flag to the GitHub Action - so it would be great to add it. If this RFE is accepted, I can also start working on it and provide a PR for it.

__Background:__
In [Eclipse BlueChi](https://github.com/eclipse-bluechi/bluechi/) we are using the multihost testing and would like to use this GH Action. But since it doesn't support enabling multihost testing yet, we have to use the testing farm cli directly.

### PR DESCRIPTION
**Pull request by engelmi**
Fixes #164 

This PR adds support for specifying pipeline settings - including the pipeline type (currently only `multihost`). The API specification of testing farm for requesting new test run is used as basis:
https://api.testing-farm.io/redoc#operation/request_a_new_test_v0_1_requests_post
(Some inline comments to the old API spec have been updated as well)

### CODE REVIEW

**Comment by engelmi** [comment]
The code coverage is failing due to the formatting. For example: 
```ts
const envSettings = envSettingsParsed.success ? envSettingsParsed.data : {};
```
Since it is in one line, the else part of the ternary operator is considered covered even though its not executed. 

The formatting of the newly added pipeline settings creates a separate line, so its considered not covered:
```ts
const pipelineSettings = pipelineSettingsParsed.success
    ? pipelineSettingsParsed.data
    : {};
```

Should I implement a new test for this or is it ok that the coverage drops a bit?

**Review by jamacku — COMMENTED** [review]
@engelmi Thank you for the PR. It looks great. Could you please also add the test into `tests/action.test.ts`?

You can take a Hardware test as an example:

https://github.com/sclorg/testing-farm-as-github-action/blob/080971e035d87c1bd99ef1235cb5a9f10ec8ae57/tests/action.test.ts#L255-L326

**Comment by engelmi** [comment]
> @engelmi Thank you for the PR. It looks great. Could you please also add the test into `tests/action.test.ts`?
 
Hi @jamacku, 
sure thing. I added two tests - valid and invalid pipeline settings. I also switched from `safeParse` to `parse` so that an error in the schema doesn't get "covered up" and shows directly before submitting the TF request.

File: tests/action.test.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
 import action from '../src/action';
 import { NewRequest } from 'testing-farm';
 import { PullRequest } from '../src/pull-request';
+import { ZodError } from 'zod';
```
Comment (jamacku) [inline tests/action.test.ts:10]:
```suggestion
```

**Review by jamacku — CHANGES_REQUESTED** [review]
Thank you for adding tests. Please remove unused import in tests and please rebase to `main` and run `yarn && yarn run all`

File: tests/action.test.ts (hunk 1 above)
Comment (engelmi) [reply tests/action.test.ts]: Done :heavy_check_mark:

**Review by jamacku — APPROVED** [review]
Thank you @engelmi, LGTM :+1:

**Pull request merged**